### ISSUE TITLE
Create a push status model

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2014]
Bodhi currently creates lock files when mashing. These are used to ensure that no two processes attempt to mash the same repository at the same time, but they are also used to store state about the current mash. Here's an example from today's ```MASHING-f27-updates``` lock file:

```
{'completed_repos': [],
 'determine_and_perform_tag_actions': True,
 'updates': ['lightdm-gtk-2.0.3-4.fc27',
             'java-9-openjdk-9.0.1.11-1.fc27',
             'icedtea-web-1.7-6.fc27',
             'kanatest-0.4.10-0.1.D20170810git19dd1a7d.fc27',
             'hexalate-1.1.1-1.fc27',
             'crossfire-client-1.72.0-1.fc27',
             'tuxmath-2.0.3-1.fc27',
             'python3-typed_ast-1.1.0-1.fc27',
             'tuxpaint-stamps-2014.08.23-1.fc27',
             'knot-resolver-1.5.0-1.fc27 knot-2.6.1-1.fc27',
             'python-stem-1.6.0-1.fc27',
             'tanglet-1.5.0-1.fc27',
             'simsu-1.3.4-1.fc27',
             'fwupd-1.0.1-3.fc27 gnome-software-3.26.2-1.fc27 libappstream-glib-0.7.4-1.fc27',
             'qt5-qtwebengine-5.9.2-2.fc27',
             'libsmbios-2.3.3-1.fc27',
             'mint-x-icons-1.4.5-1.fc27',
             'mint-y-icons-1.1.0-1.fc27',
             'pyotherside-1.5.3-2.fc27',
             'libx86emu-1.11-3.fc27 hwinfo-21.47-3.fc27']}
```

The use of these lock files limits us in a few ways. For one, it is difficult to make a good human interface to see the current mashing status (the [```/masher```](https://bodhi.fedoraproject.org/masher/) view has poor performance and is also difficult to interpret) since the lock files are only available on the backend mashing server. It is also difficult to make ```bodhi-push``` work remotely, since it tries to read these lock files.

We should refactor bodhi to store its mash state in a new ```PushStatus``` model in the database. It should be flexible enough to handle storing the current status of the mash just as the lock files do today.

This will enable further work, like creating a REST API to control mashing and ultimately web and CLI interfaces to view the current status of mashing.

**Comment by bowlofeggs** [comment]
@nirik suggested a different name, so I will go with ```ComposeStatus```. It was also suggested that we stop saying "mash", perhaps preferring "compose" from now on.

**Comment by bowlofeggs** [comment]
After realizing that I was adding a ```status``` field to the ```ComposeStatus``` class, and the silliness of there being a thing called ```ComposeStatus.status```, I decided to call the new class simply ```Compose``` so that ```Compose.status``` would be the current state of the compose. Now we're getting somewhere with this bikeshed!

**Comment by puiterwijk** [comment]
@bowlofeggs please take #1987 into account as well.
So you only need to store a single completed repo per Compose :).

**Comment by bowlofeggs** [comment]
The patch for this will be included in the upcoming 3.2.0 release:

https://github.com/fedora-infra/bodhi/pull/2092
